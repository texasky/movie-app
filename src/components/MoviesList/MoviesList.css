.movies-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    margin-bottom: 20px;
}

.movies-list-wrapper {
    max-width: 1400px;
    margin: 110px auto;
}