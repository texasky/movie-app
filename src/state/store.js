import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 