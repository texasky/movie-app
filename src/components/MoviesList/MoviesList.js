import './MoviesList.css'
import MovieItem from '../MovieItem/MovieItem';

const MoviesList = (props) => {
    return (
        <div className='movies-list'>
            {props.movies.map((movie) => (
                <MovieItem key={movie.imdbID} movie={movie}/>
            ))}
        </div>
    )
}

export default MoviesList;