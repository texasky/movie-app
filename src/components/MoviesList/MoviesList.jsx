import './MoviesList.css'
import MovieItem from '../MovieItem/MovieItem';

const MoviesList = (props) => {
    console.log(props, 'MoviesListProps')
    return (
        <div className='movies-list'>
            {props.movies.map((movie) => (
                <MovieItem key={movie.imdbID} movie={movie}/>
            ))}
        </div>
    )
}

export default MoviesList;