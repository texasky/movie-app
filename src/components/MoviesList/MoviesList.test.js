import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './../../state/store'
import MoviesList from './MoviesList';
import { BrowserRouter as Router } from 'react-router-dom';

const movies = [{
        Poster: "https://m.media-amazon.com/images/M/MV5BMGVmMWNiMDktYjQ0Mi00MWIxLTk0N2UtN2ZlYTdkN2IzNDNlXkEyXkFqcGdeQXVyODE5NzE3OTE@._V1_SX300.jpg",
        Title: "Harry Potter and the Deathly Hallows: Part 2",
        Type: "movie",
        Year: "2011",
        imdbID: "tt1201607"
    },{
        Poster: "https://m.media-amazon.com/images/M/MV5BMGVmMWNiMDktYjQ0Mi00MWIxLTk0N2UtN2ZlYTdkN2IzNDNlXkEyXkFqcGdeQXVyODE5NzE3OTE@._V1_SX300.jpg",
        Title: "Harry Potter and the Deathly Hallows: Part 3",
        Type: "movie",
        Year: "2011",
        imdbID: "tt12016072"
    }];

test('Movies list component tests', () => {
    const utils = render(<Provider store={store}><Router><MoviesList movies={movies}/></Router></Provider>);
    const input = utils.getByTestId('movie-list');
    const movieItems = utils.getAllByTestId('movie-item');

    expect(movieItems.length).toBe(2);
    expect(input).toBeInTheDocument();
});

test('Movies list component matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Router><MoviesList movies={movies}/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});