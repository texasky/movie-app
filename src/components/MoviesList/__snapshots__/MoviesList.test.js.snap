// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies list component matches snapshot 1`] = `
<div
  className="movies-list movies-list-wrapper"
  data-testid="movie-list"
>
  <div
    className="movie-item"
    data-testid="movie-item"
    onClick={[Function]}
  >
    <img
      alt="movie"
      src="https://m.media-amazon.com/images/M/MV5BMGVmMWNiMDktYjQ0Mi00MWIxLTk0N2UtN2ZlYTdkN2IzNDNlXkEyXkFqcGdeQXVyODE5NzE3OTE@._V1_SX300.jpg"
    />
    <div
      className="movie-item-title"
    >
      Harry Potter and the Deathly Hallows: Part 2
    </div>
  </div>
  <div
    className="movie-item"
    data-testid="movie-item"
    onClick={[Function]}
  >
    <img
      alt="movie"
      src="https://m.media-amazon.com/images/M/MV5BMGVmMWNiMDktYjQ0Mi00MWIxLTk0N2UtN2ZlYTdkN2IzNDNlXkEyXkFqcGdeQXVyODE5NzE3OTE@._V1_SX300.jpg"
    />
    <div
      className="movie-item-title"
    >
      Harry Potter and the Deathly Hallows: Part 3
    </div>
  </div>
</div>
`;
